azure/table_azure_application_gateway.go

```diff
@@ -0,0 +1,489 @@
+			},
+			{
+				Name:        "ssl_profiles",
+				Description: "SSL profiles of the application gateway resource.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Properties.SslProfiles"),
+			},
+			{
+				Name:        "trusted_client_certificates",
+				Description: "Trusted client certificates of the application gateway resource.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Properties.TrustedClientCertificates"),
+			},
+			{
+				Name:        "trusted_root_certificates",
+				Description: "Trusted root certificates of the application gateway resource.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Properties.TrustedRootCertificates"),
+			},
+			{
+				Name:        "url_path_maps",
+				Description: "URL path map of the application gateway resource.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Properties.URLPathMaps"),
+			},
+			{
+				Name:        "web_application_firewall_configuration",
+				Description: "Web application firewall configuration.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Properties.WebApplicationFirewallConfiguration"),
+			},
+			{
+				Name:        "zones",
+				Description: "A list of availability zones denoting where the resource needs to come from.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ApplicationGateway.Zones"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ApplicationGateway.Name"),
+			},
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ApplicationGateway.Tags"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ApplicationGateway.ID").Transform(idToAkas),
+			},
+
+			// Azure standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ApplicationGateway.Location").Transform(toLower),
+			},
+			{
+				Name:        "resource_group",
+				Description: ColumnDescriptionResourceGroup,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ApplicationGateway.ID").Transform(extractResourceGroupFromID),
+			},
+			{
+				Name:        "subscription_id",
+				Description: ColumnDescriptionSubscription,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ApplicationGateway.ID").Transform(idToSubscriptionID),
+			},
+		},
+	}
+}
+
+type ApplicationGatewayInfo struct {
+	ApplicationGateway network.ApplicationGateway
+	Properties         ApplicationGatewayPropertiesInfo
```

Comment:
do we need a custom structure?  for example - transform.fromField("ApplicationGatewayPropertiesFormat.Sku") will not work?